support/globalconfig/ingress.go

```diff
@@ -0,0 +1,38 @@
+package globalconfig
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+)
+
+func IngressConfig() *configv1.Ingress {
+	return &configv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+	}
+}
+
+func ReconcileIngressConfig(cfg *configv1.Ingress, hcp *hyperv1.HostedControlPlane) error {
+	cfg.Spec.Domain = ingressDomain(hcp)
+	globalConfig, err := ParseGlobalConfig(context.Background(), hcp.Spec.Configuration)
+	if err != nil {
+		// Cannot parse global configuration, skip picking up any settings from it
+		return fmt.Errorf("cannot parse hcp global configuration")
```

Comment:
Not getting the actual error here will be pretty frustrating :)

Reply:
technically we cannot get to this point. We validate parsing of global config in the HCP controller and skip any further reconciliation if  the config is invalid.

Reply:
refactored to not parse the global config here.